context_test.go

```diff
@@ -262,3 +263,24 @@ func TestContext_lookupFlagSet(t *testing.T) {
 		t.Fail()
 	}
 }
+
+// TestContextPropagation tests that
+// *cli.Context always has a valid
+// context.Context
+func TestContextPropagation(t *testing.T) {
+	ctx := NewContext(nil, nil, nil)
+	if ctx.Context == nil {
+		t.Fatal("expected a non nil context when no parent is present")
+	}
```

Comment:
Should be a separate test

Reply:
@AudriusButkevicius I made this into a separate test